From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Tue, 20 Mar 2018 22:07:30 +0100
Subject: Set systemwide default settings for libssl users

This config change enforeces a TLS1.2 protocol version as minimum. It
can be overwritten by the system administrator.

It also changes the default security level from 1 to 2, moving from the 80 bit
security level to the 112 bit security level.

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 apps/openssl.cnf | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/apps/openssl.cnf b/apps/openssl.cnf
index 03330e0120a2..f3bea3e61546 100644
--- a/apps/openssl.cnf
+++ b/apps/openssl.cnf
@@ -23,6 +23,9 @@ config_diagnostics = 1
 # oid_file       = $ENV::HOME/.oid
 oid_section = new_oids
 
+# System default
+openssl_conf = default_conf
+
 # To use this configuration file with the "-extfile" option of the
 # "openssl x509" utility, name here the section containing the
 # X.509v3 extensions to use:
@@ -388,3 +391,13 @@ oldcert = $insta::certout # insta.cert.pem
 # Certificate revocation
 cmd = rr
 oldcert = $insta::certout # insta.cert.pem
+
+[default_conf]
+ssl_conf = ssl_sect
+
+[ssl_sect]
+system_default = system_default_sect
+
+[system_default_sect]
+MinProtocol = TLSv1.2
+CipherString = DEFAULT@SECLEVEL=2
